theora_image_transport/CMakeLists.txt

```diff
@@ -1,79 +1,122 @@
   ${PC_THEORAENC_INCLUDE_DIRS}
-  ${PC_THEORADEC_INCLUDE_DIRS}
+  ${PC_THEORADEC_INCLUDE_DIRS})
+
+target_link_libraries(${LIBRARY_NAME}
+  ${OpenCV_LIBRARIES}
+  ${PC_OGG_LIBRARIES}
+  ${PC_THEORA_LIBRARIES}
+  ${PC_THEORAENC_LIBRARIES}
+  ${PC_THEORADEC_LIBRARIES}
 )
 
-link_directories(${PC_OGG_LIBRARY_DIRS} 
-                 ${PC_THEORA_LIBRARY_DIRS}
-                 ${PC_THEORAENC_LIBRARY_DIRS}
-                 ${PC_THEORADEC_LIBRARY_DIRS})
-add_definitions(${PC_OGG_CFLAGS_OTHER} 
-                ${PC_THEORA_CFLAGS_OTHER}
-                ${PC_THEORAENC_CFLAGS_OTHER}
-                ${PC_THEORADEC_CFLAGS_OTHER}
+ament_target_dependencies(${LIBRARY_NAME}
+  "image_transport"
+  "cv_bridge"
+  "rclcpp"
+  "pluginlib"
+  "sensor_msgs"
+  ${PROJECT_NAME}
+)
+
+install(TARGETS ${LIBRARY_NAME}
+  ARCHIVE DESTINATION lib
+  LIBRARY DESTINATION lib
+  RUNTIME DESTINATION bin
 )
 
-add_library(${PROJECT_NAME} src/theora_publisher.cpp src/theora_subscriber.cpp src/manifest.cpp)
-add_dependencies(${PROJECT_NAME} ${PROJECT_NAME}_gencfg ${PROJECT_NAME}_gencpp)
-target_link_libraries(${PROJECT_NAME} ${catkin_LIBRARIES}
-                                      ${OpenCV_LIBRARIES}
-                                      ${PC_OGG_LIBRARIES}
-                                      ${PC_THEORA_LIBRARIES}
-                                      ${PC_THEORAENC_LIBRARIES}
-                                      ${PC_THEORADEC_LIBRARIES}
+install(
+  DIRECTORY "include/"
+  DESTINATION include
 )
 
-class_loader_hide_library_symbols(${PROJECT_NAME})
+add_definitions(${PC_OGG_CFLAGS_OTHER}
+                ${PC_THEORA_CFLAGS_OTHER}
+                ${PC_THEORAENC_CFLAGS_OTHER}
+                ${PC_THEORADEC_CFLAGS_OTHER})
 
 add_executable(ogg_saver src/ogg_saver.cpp)
 target_link_libraries(ogg_saver ${PC_THEORA_LIBRARY} 
                                 ${PC_OGG_LIBRARY} 
                                 ${OpenCV_LIBRARIES} 
-                                ${catkin_LIBRARIES}  
                                 ${PC_THEORAENC_LIBRARIES}
                                 ${PC_THEORADEC_LIBRARIES})
-add_dependencies(ogg_saver ${PROJECT_NAME}_gencpp)
 
-install(TARGETS ${PROJECT_NAME} ogg_saver
-  ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-  LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-  RUNTIME DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
+ament_target_dependencies(ogg_saver
+  "rclcpp"
+  ${PROJECT_NAME})
+
+rosidl_generate_interfaces(${PROJECT_NAME}
+  "msg/Packet.msg"
+  DEPENDENCIES
+    builtin_interfaces
+    std_msgs
+    sensor_msgs
+   ADD_LINTER_TESTS
 )
-install(DIRECTORY include/${PROJECT_NAME}/
-  DESTINATION ${CATKIN_PACKAGE_INCLUDE_DESTINATION}
+
+rosidl_target_interfaces(${LIBRARY_NAME}
+  ${PROJECT_NAME} "rosidl_typesupport_cpp") 
+
+rosidl_target_interfaces(ogg_saver
+  ${PROJECT_NAME} "rosidl_typesupport_cpp") 
+
+ament_export_dependencies(rosidl_default_runtime)
+
+install(TARGETS ${LIBRARY_NAME} ogg_saver
```

Comment:
if `ogg_saver` is an executable that we expect to run using `ros2 run`, we should install it in to `lib/${PROJECT_NAME}`